shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java

```diff
@@ -181,11 +184,30 @@ public FlutterEngine getFlutterEngine() {
   public void onAttach(Context context) {
     super.onAttach(context);
 
+    initializeFlutter(getContextCompat());
+
     // When "retain instance" is true, the FlutterEngine will survive configuration
     // changes. Therefore, we create a new one only if one does not already exist.
     if (flutterEngine == null) {
       createFlutterEngine();
     }
+
+    // Regardless of whether or not a FlutterEngine already existed, the PlatformPlugin
+    // is bound to a specific Activity. Therefore, it needs to be created and configured
+    // every time this Fragment attaches to a new Activity.
+    // TODO(mattcarroll): the PlatformPlugin needs to be reimagined because it implicitly takes
+    //                    control of the entire window. This is unacceptable for non-fullscreen
+    //                    use-cases.
+    platformPlugin = new PlatformPlugin(getActivity(), flutterEngine.getPlatformChannel());
```

Comment:
The way this is handled in the iOS embedding is to new this up whenever the Engine actually gets a ViewController.  Could that work here?  Perhaps as an else on your `if (flutterEngine == null) { ... } else { flutterEngine.notifyContextAttached(context); }`?  I assume the context is acceptable here instead of activity, but if not just pass through getActivity.  That way the engine manages the plugin instead of the view.

Reply:
What is the problem that you're attempting to solve with this suggestion?

Reply:
The fragment doesn't need to get involved in any of this.  Make the engine manage the PlatformPlugin, and that way you can keep the abstraction more contained and replace it more easily.

Reply:
Please notice the constructor - it requires an Activity.  This plugin is a problem for multiple reasons.  But one thing that's for sure is that a FlutterEngine should not be taking references to an Activity to then perform behaviors on the Activity.  That's the opposite direction of this refactoring effort.  

I don't really want to focus on this issue here because the purpose of the TODOs were to indicate that I need to look deeper at these problems, because the solution isn't obvious.  If you'd like to chat about the eventual solution, we can, but I think these TODO issues go beyond the scope of this specific PR.